caikit/interfaces/common/data_model/remote.py

```diff
@@ -0,0 +1,139 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This file contains interfaces required to connect to Remote servers
+"""
+
+# Standard
+from dataclasses import field
+from http.client import HTTP_PORT, HTTPS_PORT
+from pathlib import Path
+from typing import Optional
+
+# First Party
+from py_to_proto.dataclass_to_proto import Annotated, Dict, FieldNumber
+import alog
+
+# Local
+from caikit.core.data_model import PACKAGE_COMMON, DataObjectBase, dataobject
+from caikit.core.exceptions import error_handler
+
+log = alog.use_channel("CNNCTDM")
+error = error_handler.get(log)
+
+
+@dataobject(PACKAGE_COMMON)
+class ConnectionTlsInfo(DataObjectBase):
+    """Helper dataclass to store information regarding TLS information."""
+
+    enabled: Annotated[bool, FieldNumber(1)] = False
+    ca_file: Annotated[Optional[str], FieldNumber(2)]
+    cert_file: Annotated[Optional[str], FieldNumber(3)]
+    key_file: Annotated[Optional[str], FieldNumber(4)]
+    insecure_verify: Annotated[bool, FieldNumber(5)] = False
+
+    # Helper variables to store the read data from TLS files
+    ca_file_data: Optional[bytes] = None
+    cert_file_data: Optional[bytes] = None
+    key_file_data: Optional[bytes] = None
```

Comment:
@gabe-l-hart this was originally kept separate because GRPC wanted bytes while requests wanted the string name. I think `requests.Session` supports bytes inputs so I think option 1 is the cleanest

Reply:
Upon further investigation it looks like HTTP requires a path while GRPC requires bytes. I propose renaming these fields to remove any suffix e.g. `ca`,`cert`,`key` and then automatically filling in the "views" with suffixes e.g. `ca_file`, `ca_data`, `cert_file`, .... and so on

Reply:
Actually after starting to implement the above I actually think file paths should be the only supported arguments. Otherwise Caikit has to handle creating a tempfile and managing its lifecycle. IIRC all of our tests/users already use TLS paths so I think that's fine.